packages/protocol/contracts/stability/Reserve.sol

```diff
@@ -345,11 +359,8 @@ contract Reserve is
    * @param value The amount of gold to transfer.
    * @return Returns true if the transaction succeeds.
    */
-  function transferExchangeGold(address payable to, uint256 value)
-    external
-    onlyRegisteredContract(EXCHANGE_REGISTRY_ID)
-    returns (bool)
-  {
+  function transferExchangeGold(address payable to, uint256 value) external returns (bool) {
```

Comment:
Have you thought about the deployment strategy for this change? I.e. to bundle the addition of the exchangeSpender proposal?

Reply:
Yes, it should be included in the proposal. Shall I document this somewhere?

Reply:
That would be a good idea I think. In the absence of an actual process (which we should have), I'd recommend the PR description for now

Reply:
could we continue to allow the registered exchange as a spender, rather than having to add them separately? then init data becomes less crucial

Reply:
so the one that is already in the registry would be always an allowed spender? That works, but when we deploy another stable asset then we will have the same problem (although less critical, because it can be frozen or oracles not reporting at the time of activation).

Reply:
I think @yorhodes is saying that we should special case the current exchange, but all future ones should be added through your function. I agree with @yorhodes

Reply:
And how would that look like in the (very unlikely) event of cUSD exchange removal?

Reply:
I'd say let's cross that bridge when we get there :P

In all seriousness, a contract upgrade or pointing the registry entry to 0x0 seem possible. But as you said, seems incredibly unlikely

Reply:
I still don't follow what @yorhodes is proposing, how would that look klike in this PR?

Reply:
You could have an `function isAllowedToSpend` that does `return isExchangeSpender.isSpender[msg.sender] || registry.getAddressForOrDie(EXCHANGE_REGISTRY_ID) == msg.sender`

Reply:
done

Reply:
When you say done, is it in this PR (I dont see it)

Reply:
damn, forgot to push